src/VisualStudio/IntegrationTest/New.IntegrationTests/CSharp/CSharpGoToImplementation.cs

```diff
@@ -91,14 +154,46 @@ public void SomeMethod()
             await TestServices.Editor.GoToDefinitionAsync(HangMitigatingCancellationToken);
             Assert.Equal("IDisposable [from metadata]", await TestServices.Shell.GetActiveWindowCaptionAsync(HangMitigatingCancellationToken));
             await TestServices.Editor.GoToImplementationAsync(HangMitigatingCancellationToken);
+
+            string identifierWithCaret;
+            if (!asyncNavigation)
+            {
+                // The navigation completed synchronously; no further action necessary
+                identifierWithCaret = "Implementation$$";
```

Comment:
Hmm, so sync navigation puts the caret in one place, and invoking the navigated item puts in a different place? Not crtiical but @CyrusNajmabadi is this intentional?